eng/pipelines/templates/jobs/ci.yml

```diff
@@ -345,9 +349,24 @@ jobs:
             Directory: sdk/${{ parameters.ServiceDirectory }}
           CheckLinkGuidance: $true
 
-      - template: /eng/common/pipelines/templates/steps/verify-samples.yml
-        parameters:
-          ServiceDirectory: ${{ parameters.ServiceDirectory }}
+      - ${{ if and(eq(variables['Build.Reason'], 'PullRequest'), eq(parameters.ServiceDirectory, 'auto')) }}:
```

Comment:
Similar heard I would like to avoid unnecessary forks in logic if we can make this work for both cases with PackageInfos.